### ISSUE TITLE
asynchronous suspension... HiveError: This should not happen. Please open an issue on GitHub.

### ISSUE THREAD

**Issue by uchihasharukh** [hivedb/hive#803]
**Steps to Reproduce**
Please tell me exactly how to reproduce the problem you are running into.

**Code sample**
```
 try {
    final FlutterSecureStorage secureStorage = const FlutterSecureStorage();
    var key = Hive.generateSecureKey();
    var encryptedBox = await Hive.openBox('tokenBox', encryptionCipher: HiveAesCipher(key));
    encryptedBox.put('token', token);
    await secureStorage.write(
      key: "${EnvironmentConfig.STORAGE_KEY}",
      value: json.encode(key),
    );
    encryptedBox.close();
  } catch (e) {
    throw (e.toString());
  }
```
**Logs**
```
E/flutter ( 5836): [ERROR:flutter/lib/ui/ui_dart_state.cc(199)] Unhandled Exception: HiveError: This should not happen. Please open an issue on GitHub.
E/flutter ( 5836): #0      storeToken
E/flutter ( 5836): <asynchronous suspension>
E/flutter ( 5836): #1      _TestScaffoldState.build.<anonymous closure>
E/flutter ( 5836): <asynchronous suspension>
E/flutter ( 5836):
E/flutter ( 5836): [ERROR:flutter/lib/ui/ui_dart_state.cc(199)] Unhandled Exception: HiveError: This should not happen. Please open an issue on GitHub.
E/flutter ( 5836): #0      BinaryReaderImpl.readFrame
E/flutter ( 5836): #1      FrameHelper.framesFromBytes
E/flutter ( 5836): #2      FrameIoHelper.framesFromFile
E/flutter ( 5836): <asynchronous suspension>
E/flutter ( 5836): #3      StorageBackendVm.initialize
E/flutter ( 5836): <asynchronous suspension>
E/flutter ( 5836): #4      HiveImpl._openBox
E/flutter ( 5836): <asynchronous suspension>
E/flutter ( 5836): #5      HiveImpl.openBox
E/flutter ( 5836): <asynchronous suspension>
E/flutter ( 5836): #6      storeToken
E/flutter ( 5836): <asynchronous suspension>
E/flutter ( 5836): #7      _TestScaffoldState.build.<anonymous closure>
E/flutter ( 5836): <asynchronous suspension>
E/flutter ( 5836):

```
**Version**
 - Platform: android,ios
 - Flutter version: 2.2.3
  hive: ^2.0.4
  hive_flutter: ^1.1.0

**Comment by uchihasharukh** [comment]
hey guys any fix done?? or is there any way to encrypt storage in flutter.????

**Comment by themisir** [comment]
Closed with #914.

Please upgrade to hive [v2.0.6](https://pub.dev/packages/hive/versions/2.0.6).
